build.gradle

```diff
@@ -458,3 +458,22 @@ project(':iceberg-spark-runtime') {
   }
 }
 
+String getProjectVersion() {
+  try {
+    return gitVersion()
+  } catch (NullPointerException e) {
+    def noGitVersion = '0.1.0-SNAPSHOT'
+    println "no .git directory available, use version $noGitVersion"
+    return noGitVersion
+  }
+}
+
+String getBranchName() {
+  try {
+    return versionDetails().branchName
+  } catch (NullPointerException e) {
+    def branchName = 'noGitBranch'
```

Comment:
I think this should default to version, since we use this to override version in javadocs.